### ISSUE TITLE
Error on `pod install` after upgrade to 0.36.0.rc.1

### ISSUE THREAD

**Issue by sstadelman** [CocoaPods/CocoaPods#3195]
I've been successfully using a Podfile referencing a local pod, using 0.35.  Immediately on upgrade to 0.36.0.rc.1, I am getting a new `undefined method` error in Xcodeproj on `pod install`.

I was originally linking with two targets.  Since the error referenced an array, I tried cutting to just one target, but this had no effect.  I suspect that this is related to the issue discussed here:  https://github.com/CocoaPods/CocoaPods/issues/3091.  If I go into the pod spec, and comment out the `s.resources` references (which includes the contents of bundles), then I can run `pod install` successfully.

*added comment on `s.resources` to resolve issue*
```
  s.library      = 'stdc++.6.0.9', 'z', 'c++'
  s.framework   = 'CoreFoundation', 'Security', 'CFNetwork', 'MobileCoreServices', 'SystemConfiguration', 'MessageUI', 'CoreData'
  s.source_files = "MAFReuse/iOS/includes/public/MAFUIComponents/*.h", "MAFReuse/iOS/includes/public/MAFLogViewer/*.h", "ODataFramework/iOS/includes/public/**/*.h"

  # s.resources = 'MAFReuse/iOS/includes/bundles/MAFUIComponents.bundle', 'ODataFramework/iOS/includes/bundles/*.bundle', 'ODataFramework/iOS/includes/bundles/*.bundle', 'ODataFramework/iOS/includes/bundles/MobilePlace.bundle/Base.lproj/MobilePlace_iPhone.storyboard'
```

(reverting to 0.35 also worked).  Here's the console:

### Command

```
/usr/local/bin/pod install
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 0.36.0.rc.1
        Ruby : ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin14.0]
    RubyGems : 2.2.2
        Host : Mac OS X 10.10.2 (14C109)
       Xcode : 6.3 (6D520o)
         Git : git version 2.1.2
Ruby lib dir : /usr/local/Cellar/ruby/2.1.3_1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ bb653fb17a2d69ed808c60eab389d29ad9b9dd98
```

### Plugins

```
cocoapods-plugins : 0.4.1
cocoapods-trunk   : 0.5.1
cocoapods-try     : 0.4.3
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
platform :ios, '8.0'

# Requires Cocoapods ruby gem (www.cocoapods.org)
# To install, run `$ sudo gem install cocoapods`

# For setup with SAP Mobile SDK:
# (1) Alter the :path value for 'NativeSDK', to match the location of the "NativeSDK" folder in your SAP Mobile SDK installation directory.
# (2) `cd` to the project folder, and run `pod install`, to setup the NativeSDK pod with your project.

target 'testNewCP' do
pod 'NativeSDK', :path => "~/SAP/MobileSDK3/NativeSDK/"
end
```

### Error

```
NoMethodError - undefined method `[]' for #<Xcodeproj::Project::Object::PBXFileReference:0x007fcf522734a8>
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/project.rb:186:in `block in add_file_reference'
/usr/local/Cellar/ruby/2.1.3_1/lib/ruby/2.1.0/pathname.rb:266:in `block in each_filename'
/usr/local/Cellar/ruby/2.1.3_1/lib/ruby/2.1.0/pathname.rb:266:in `each'
/usr/local/Cellar/ruby/2.1.3_1/lib/ruby/2.1.0/pathname.rb:266:in `each_filename'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/project.rb:184:in `add_file_reference'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:168:in `block (2 levels) in add_file_accessors_paths_to_pods_group'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:166:in `each'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:166:in `block in add_file_accessors_paths_to_pods_group'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:162:in `each'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:162:in `add_file_accessors_paths_to_pods_group'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:103:in `block in add_resources'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/user_interface.rb:110:in `message'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:102:in `add_resources'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer/file_references_installer.rb:38:in `install!'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer.rb:486:in `install_file_references'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer.rb:130:in `block in generate_pods_project'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/user_interface.rb:49:in `section'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer.rb:128:in `generate_pods_project'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/installer.rb:96:in `install!'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/command/project.rb:101:in `run'
/usr/local/lib/ruby/gems/2.1.0/gems/claide-0.8.1/lib/claide/command.rb:312:in `run'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/lib/cocoapods/command.rb:46:in `run'
/usr/local/lib/ruby/gems/2.1.0/gems/cocoapods-0.36.0.rc.1/bin/pod:44:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by cooler333** [comment]
same issue

**Comment by pj4533** [comment]
same issue here.....something with resource definitions in pod specs?

**Comment by cooler333** [comment]
@pj4533 i'm get this error because of image.xcassets file. Just delete it from resource

**Comment by pj4533** [comment]
yeah I don't know, did't have time to dig in too much...I just downgraded to 0.35 and its working fine.

**Comment by lazerwalker** [comment]
I'm experiencing the same issue with IntentKit:

* [Pod install stacktrace](https://github.com/intentkit/IntentKit/issues/51)
* [Test project Podfile](https://github.com/intentkit/IntentKit/blob/master/Podfile),
* [Podspec](https://github.com/intentkit/IntentKit/blob/master/IntentKit.podspec)

**Comment by ohwutup** [comment]
+1

**Comment by kylef** [comment]
This is caused when you have the same resource included multiple times:

```ruby
s.resources = 'MAFReuse/iOS/includes/bundles/MAFUIComponents.bundle', 'ODataFramework/iOS/includes/bundles/*.bundle', 'ODataFramework/iOS/includes/bundles/*.bundle', 'ODataFramework/iOS/includes/bundles/MobilePlace.bundle/Base.lproj/MobilePlace_iPhone.storyboard'
```

Notice, you have `ODataFramework/iOS/includes/bundles/*.bundle` listed in the resources twice.

**Comment by sstadelman** [comment]
@kylef nice find.  Did notice eventually, but didn't occur to me that they might be related.  BR/Stan

**Comment by caseysackett** [comment]
+1

**Comment by ohwutup** [comment]
i think there is some more going on with this, i have removed all dupes from my `resources`, `source_files`, and `public_header_files` and the issue persists

**Comment by gauravthummar** [comment]
There is nothing duplicate in my code. i just create fresh project. and also delete the image.xcassets file. but still it give same error. nothing workout.

**Comment by segiddins** [comment]
This should be fixed in the 0.38 beta?

**Comment by mave99a** [comment]
Still same problem in 0.38 beta :(

**Comment by segiddins** [comment]
This is due to reflecting file system structure for development pods, if someone wants to try and dig into fixing it.

**Comment by segiddins** [comment]
This should be fixed as of 1.0, due to @efirestone's improved handling of resources.

**Comment by efirestone** [comment]
Yup! To be clear, use a resources glob of something like `Resources/**/*` and it will handle everything inside properly now.
